shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -152,11 +152,12 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
 }
 
 std::vector<std::shared_ptr<FlutterPlatformViewLayer>>
-FlutterPlatformViewLayerPool::GetUnusedLayers() {
+FlutterPlatformViewLayerPool::RemoveUnusedLayers() {
   std::vector<std::shared_ptr<FlutterPlatformViewLayer>> results;
   for (size_t i = available_layer_index_; i < layers_.size(); i++) {
     results.push_back(layers_[i]);
   }
+  layers_.erase(layers_.begin() + available_layer_index_, layers_.end());
```

Comment:
Does it defeat the purpose of having a layer "pool"? Should we just remove the pool? 

On a side note, I recently plan to remove the mask pool, since it won't be too helpful as we now use CALayer as the mask. It also makes it easier for me to remove the UIView as the mask.

Reply:
No, because the layer pool holds onto layers that are in use between frames. Something needs to hold onto this data.

Reply:
I meant, to hold onto this data, we just need the `layers_` vector. Do we still need this "pool", which is mainly created for recycling purpose (if im reading it correctly)

Reply:
We'd also need to keep the `available_layer_index_` around, but since its only purpose is to keep the position of the next available layer, it's kind of nice to have the two bundled underneath an interface, which also serves to make the code a bit more readable than straight vector manipulation.

Reply:
Oh is the new code still recycling the layers? I may have missed it.